### PR TITLE
#28 policy: milestone-first triage for agent workflows

### LINKED ISSUE #28: policy: milestone-first triage for agent workflows
## Context
We want vibe-driven work to prioritize milestones so execution stays aligned with shipping goals and avoids scope drift.

## Goal
Enforce milestone-first triage policy for agent workflows.

## Scope (MVP)
- Policy behavior:
  - prioritize current/most urgent milestone before backlog from future milestones
  - if user asks for unrelated new work, default to creating a new issue in a future milestone unless user explicitly asks to implement immediately
- Integrate policy into agent-facing workflow/docs (`AGENTS.md` + generated guidance in `.vibe` templates if applicable).
- Optional CLI support:
  - warning in triage/status when proposed work is outside active milestone focus.

## Scope extension (2026-02-16)
- Add pre-merge review suggestion policy:
  - after implementation and before merge, agent should suggest running a code review step (e.g. Codex `/review`)
  - keep suggestion explicit in final handoff/checklist
  - if review is skipped, require explicit user acknowledgment in workflow notes

## DoD
- policy documented in agent workflow contract
- issue triage prompts reflect milestone-first defaults
- pre-merge review suggestion is included in completion workflow
- tests/docs updated where behavior is automated
- validation: `pnpm test` + `pnpm build`

### PR DESCRIPTION
**Pull request by lucasgday**
## Summary
- Issue: #28 policy: milestone-first triage for agent workflows
- Branch: `issue-28-semantic-milestones`
- Issue URL: https://github.com/lucasgday/vibe-backlog/issues/28

## Architecture decisions
- Keep work scoped to issue #28 on branch `issue-28-semantic-milestones`.
- Preserve existing CLI command contracts; harden behavior without changing user-facing flags.
- Use deterministic git/GitHub state and marker-based outputs to keep runs reproducible.

## Why these decisions were made
- Avoid placeholder rationale in PRs so reviewers always have actionable context.
- Persist review artifacts in one final commit/push to guarantee traceable run state.
- Stop retry loops without progress to reduce wasted review cycles and noisy follow-ups.

## Alternatives considered / rejected
- Keep placeholder rationale and rely on manual cleanup: rejected due to inconsistent PR quality.
- Retry until max attempts regardless of autofix progress: rejected due to repeated no-op loops.
- Commit on every attempt: rejected due to noisy history and harder rollback auditing.

Fixes #28

### CODE REVIEW

File: tests/tracker.test.ts [hunk 1]
```diff
@@ -364,6 +377,70 @@ describe.sequential("tracker reconcile core", () => {
...
+        isInteractive: false,
+      },
+    );
+
+    expect(result.issueUpdates).toHaveLength(1);
+    expect(result.issueUpdates[0]?.milestoneSource).toBe("generated");
+    expect(result.commands.some((args) => args[0] === "api" && args[1] === "--method" && args[2] === "POST")).toBe(true);
+  });
+
+  it("reuses existing milestone when generated title normalizes to existing one", async () => {
```
Comment (lucasgday) [inline tests/tracker.test.ts]: **[P2] Missing regression test for whitespace-normalized milestone reuse**

Tests validate case-insensitive reuse but do not cover internal-whitespace normalization (for example, `Billing:  Retry   Hardening`). Without this case, duplicate-creation regressions in normalization logic can pass CI.

Pass: `quality`

File: src/cli-program.ts [hunk 2]
```diff
@@ -394,6 +393,47 @@ function printHygieneWarnings(snapshots: IssueSnapshot[]): void {
   }
 }
 
+async function printPreflightMilestoneSuggestions(execaFn: ExecaFn, snapshots: IssueSnapshot[]): Promise<void> {
+  const missingMilestone = findIssuesMissingMilestone(snapshots);
+  if (!missingMilestone.length) return;
+
+  try {
+    const reconcilePlan = await runTrackerReconcile(
```
Comment (lucasgday) [inline src/cli-program.ts]: **[P2] Preflight now triggers full-repo reconcile reads**

`preflight` now calls `runTrackerReconcile({dryRun:true})`, which fetches repository labels, milestones, and issues with `state=all`. This adds heavy API traffic and latency to the default read-only command path and can degrade reliability on repos with large issue history.

Pass: `ops`

**Review by lucasgday — COMMENTED** [review]
vibe review: final report posted.
